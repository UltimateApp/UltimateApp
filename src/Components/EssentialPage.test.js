import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../Store/testStore';

import ConectedEssentialPage, { EssentialPage } from './EssentialPage';

beforeEach(() => jest.useFakeTimers()); // for Modal behaviour

describe('<EssentialPage />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConectedEssentialPage />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
