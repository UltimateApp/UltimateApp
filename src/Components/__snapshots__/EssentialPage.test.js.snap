// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EssentialPage /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 5,
            "height": "70%",
            "justifyContent": "space-around",
            "margin": 20,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "80%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#e5e5e5",
              "borderColor": "#232b2b",
              "borderRadius": 2,
              "borderWidth": 1,
              "elevation": 2,
              "marginHorizontal": 5,
              "padding": 10,
              "paddingHorizontal": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 1.41,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#232b2b",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Cutting
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#232b2b",
              "borderRadius": 10,
              "elevation": 2,
              "padding": 10,
              "paddingHorizontal": 20,
              "width": 120,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingTop": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "elevation": 2,
          "padding": 10,
          "paddingVertical": 10,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 1.41,
          "width": "50%",
        }
      }
    >
      <View
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#232b2b",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Cutting
          </Text>
          <Text />
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#e5e5e5",
              "borderBottomWidth": 1,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingTop": 20,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "paddingBottom": 50,
              "paddingVertical": 10,
            }
          }
          data={
            Array [
              Object {
                "animation": "",
                "id": 1,
                "text": "Rise Up Ultimate",
                "title": "How to make a perfect cut?",
                "video": "424000350",
              },
            ]
          }
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "flex": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "paddingVertical": 15,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "paddingLeft": 10,
                      }
                    }
                  >
                    How to make a perfect cut?
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "width": 750,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={null}
                    >
                      <View
                        accessibilityIgnoresInvertColors={true}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "rgb(0,0,0)",
                            "borderRadius": 5,
                            "height": 250,
                            "justifyContent": "center",
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "uri": undefined,
                            }
                          }
                          style={
                            Array [
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              },
                              Object {
                                "height": 250,
                                "width": "100%",
                              },
                              undefined,
                            ]
                          }
                        />
                        <View
                          style={
                            Object {
                              "backgroundColor": "#232b2b",
                              "bottom": 10,
                              "paddingHorizontal": 5,
                              "position": "absolute",
                              "right": 5,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#fff",
                                "fontSize": 16,
                                "textAlign": "center",
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "paddingVertical": 15,
                        "textAlign": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 14,
                          "paddingLeft": 10,
                        }
                      }
                    >
                      Rise Up Ultimate
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "borderBottomColor": "#e5e5e5",
                        "borderBottomWidth": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
        <Text
          style={
            Object {
              "height": 30,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
